connectors/es/index.py

```diff
@@ -64,8 +62,8 @@ async def fetch_response_by_id(self, doc_id):
 
         return resp.body
 
-    async def index(self, doc):
-        return await self.client.index(index=self.index_name, document=doc)
+    async def index(self, doc, doc_id=None):
+        return await self.client.index(index=self.index_name, id=doc_id, document=doc)
```

Comment:
What about this change? Why is doc_id added?

Reply:
because serverless does not support `POST /index/_doc` anymore to create a new doc, we want `POST /index/_doc/<new-id>`

Reply:
In that case should `doc_id` be a required param, and not have a default value of `None`?

Reply:
it was a mistake from serverless gating, I will revert that once they push https://github.com/elastic/elasticsearch/pull/96752